extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +147,181 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   */
+  processCrossDomainIframe(win, frameUrl) {
```

Comment:
nit: i feel iframe related logic deserves a separate file e.g.: iframe-transport.js
to not bloat the test file